src/tools/pkcs11-tool.c

```diff
@@ -3153,9 +3153,63 @@ static int write_object(CK_SESSION_HANDLE session)
 		}
 #endif
 #endif
-	}
-	else
-	if (opt_object_class == CKO_DATA) {
+		break;
+	case CKO_SECRET_KEY:
+		clazz = CKO_SECRET_KEY;
+		type = CKK_AES;
+
+		if (opt_key_type != 0) {
+			if (strncmp(opt_key_type, "AES:", strlen("AES:")) == 0 || strncmp(opt_key_type, "aes:", strlen("aes:")) == 0)
+				type = CKK_AES;
+			else if (strncmp(opt_key_type, "DES:", strlen("DES:")) == 0 || strncmp(opt_key_type, "des:", strlen("des:")) == 0)
```

Comment:
Not sure if any new code should be added for DES (as in single key DES) whereas there are plenty of systems that make use of 2 key 3DES (and would be nice to have explicit support for this in import level, without having to do manual key shuffling before)

Reply:
I think this option list should to be equal whith generate key options.
https://github.com/OpenSC/OpenSC/blob/master/src/tools/pkcs11-tool.c#L2460
Or will I remove this option from gen_key also?